pyleros/alu.py

```diff
@@ -0,0 +1,72 @@
+from myhdl import instances, block, intbv, \
+                    always_comb, always_seq
+
+from pyleros.types import alu_op_type, t_decSignal
+
+
+
+@block
+def pyleros_alu(dec, acc, opd, pre_acc):
+    """The alu module for pyleros, purely combinatorial
+
+    Arguments (ports):
+        dec: IN The decoder control signals
+        acc: IN The accumulator
+        opd: IN Operand, based on whether the instruction is 
+            immediate or not
+        pre_acc: OUT The result of the ALU operation, from the acc mux
+
+    Parameters:
+        None
+    """
+
+    res_arith, res_log = intbv(0)[16:], intbv(0)[16:]
+
+    # Add and Subtract module
+    @always_comb
+    def op_add_sub():
+
+        if not dec[int(t_decSignal.add_sub)]:
```

Comment:
I need to connect multiple control signals(around 10) from the decoder to the fetch, execute, alu, etc. The alternative to creating the list was to add about 10 signals to the argument list in each of these modules, and their instantiations. I've used this right now to make the code shorter.  The other option I can see is creating an interface?

What would you suggest?